captum/optim/_utils/models.py

```diff
@@ -151,3 +155,106 @@ def forward(self, x: torch.Tensor) -> torch.Tensor:
             self.dilation,
             self.groups,
         )
+
+
+def pad_reflective_a4d(x: torch.Tensor, padding: List[int]) -> torch.Tensor:
+    """
+    Reflective padding for all 4 dimensions of an NCHW tensor
+    """
+
+    assert x.dim() == 4
+    assert len(padding) == 8
+
+    if x.size(0) == 2:
+        assert sum(padding[6:]) == 0
+    if x.size(1) == 2:
+        assert sum(padding[4:6]) == 0
+

```

Comment:
Long time ago I implemented circular padding in PT, I was wondering if we can adopt similar solution here so that we don't have to have if-elif for x.size(0).

https://github.com/NarineK/pytorch/blob/0a877cae6bf8e665b104d1c5a36c2a1c60f14a44/torch/nn/functional.py#L3020
I can try to adopt that solution for reflective padding.

Reply:
There is in fact a more elegant solution with slice but it is not supported w/torch script
https://github.com/pytorch/pytorch/commit/6832149459c698ca2460670c6eb46e5948262e49#diff-c358c214f663ba0c8b9c6846fbe0042fa29494cf02fe4714a17dcd0d268b035bR3011

Reply:
@NarineK That be really awesome to not have the limitation! Are you able to implement it yourself?

Reply:
@NarineK Actually, I just got a reply back from Chris:

> Reflection would introduce artifacts (encourage the image to loop around at boundaries). You could do it, but I'd recommend against.
> Constant padding wouldn't be the end of the world.

So, it looks like we need symmetric to avoid artifacts.

Reply:
Thank you for sharing the feedback, @ProGamerGov!

I'd actually recommend to convert the tensor into numpy, perform the padding in numpy and convert it back into tensor. You can do it by updating the data of the tensor and keeping other properties of the tensor such as `requires_grad`.
`tens.data` will give you the data and you can call numpy padding on it. 
It would make sense to make padding techniques in PT more sophisticated. It looks like currently the padding doesn't work when dim_size <= padding. We need to fix that in PyTorch.

Reply:
`tens.data.cpu().numpy()` in case the tensor is on a GPU device.

Reply:
@NarineK Thanks for the idea! I was able to get it working with `torch.autograd.Function`, so SharedImage now uses NumPy's symmetric padding with preserving the gradient!